### PR TITLE
Expand docs

### PR DESCRIPTION
**Pull request by hdgarrood**
* Say what the function is/does before talking about why you might want to use it
* Update the reference to `runConj` (which no longer exists)
* Be more specific than "for free". I've grown to dislike the phrase "for free" because it can mean too many different things in different contexts, and it's usually clearer to just say what you really mean.

### CODE REVIEW

File: src/Safe/Coerce.purs [hunk 1]
```diff
@@ -6,11 +6,18 @@ module Safe.Coerce
...
+-- | `unsafeCoerce` at runtime. Unlike `unsafeCoerce`, it is safe, because the
+-- | `Coercible` constraint prevents any use of this function from compiling
+-- | unless the compiler can prove that the two types have the same runtime
+-- | representation.
+-- |
 -- | One application for this function is to avoid doing work that you know is a
 -- | no-op because of newtypes. For example, if you have an `Array (Conj a)` and you
--- | want an `Array (Disj a)`, you could do `Data.Array.map (runConj >>> Disj)`, but
+-- | want an `Array (Disj a)`, you could do `Data.Array.map (un Conj >>> Disj)`, but
 -- | this performs an unnecessary traversal. `coerce` accomplishes the same
```
Comment (JordanMartinez) [inline src/Safe/Coerce.purs:19]: Would it help to note that the unnecessary traversal in this case is `O(n)` to help show the contrast more?

**Review by JordanMartinez — APPROVED** [review]
This is definitely cleaner. LGTM!

**Comment by JordanMartinez** [comment]
Hm... CI is failing because there isn't a test folder and its running `pulp test`.

File: src/Safe/Coerce.purs (hunk 1 above)
Comment (hdgarrood) [reply src/Safe/Coerce.purs:19]: Yes, that sounds good to me

**Pull request merged**